src/ColorVectorSpace.jl

```diff
@@ -311,38 +311,115 @@ typemax(::Type{T}) where {T<:ColorTypes.AbstractGray} = T(typemax(eltype(T)))
 typemin(::T) where {T<:ColorTypes.AbstractGray} = T(typemin(eltype(T)))
 typemax(::T) where {T<:ColorTypes.AbstractGray} = T(typemax(eltype(T)))
 
-include("precompile.jl")
-_precompile_()
+## RGB tensor products
+
+"""
+    RGBRGB(rr, gr, br, rg, gg, bg, rb, gb, bb)
+
+Represent the [tensor product](https://en.wikipedia.org/wiki/Tensor_product) of two RGB values.
+
+# Example
+
+```jldoctest
+julia> a, b = RGB(0.2f0, 0.3f0, 0.5f0), RGB(0.77f0, 0.11f0, 0.22f0)
+(RGB{Float32}(0.2f0,0.3f0,0.5f0), RGB{Float32}(0.77f0,0.11f0,0.22f0))
+
+julia> a ⊗ b
+RGBRGB{Float32}(
+ 0.154f0  0.022f0  0.044f0
+ 0.231f0  0.033f0  0.066f0
+ 0.385f0  0.055f0  0.11f0 )
+"""
+struct RGBRGB{T}
+    rr::T
+    gr::T
+    br::T
+    rg::T
+    gg::T
+    bg::T
+    rb::T
+    gb::T
+    bb::T
+end
```

Comment:
TBH, I'm not sure how useful the tensor product for colors is...:sweat_smile:

If we introduce new types for the tensor product, I think it's better to make them parametric. Even if they are non-parametric, they should have a parametric supertype.

The reason is that we may also need the tensor product types for color types other than `RGB` (e.g. `ARGB`, `RGBA`, `AGray` and `GrayA`) for consistency rather than practicality. Probably Colors.jl will not support tensor products unless TensorCore is in the `Base` or stdlib. In that case, `XYZ` and `LMS` may need to be extended within ColorVectorSpace.

Also, If we introduce new types, I think it's better to do lazy evaluation. On modern CPUs, the bottleneck is often memory access rather than multiplication.

Creating such a gadget only for color types is an over-investment, but I think doing it in TensorCore.jl isn't too bad.

Reply:
> TBH, I'm not sure how useful the tensor product for colors is...sweat_smile

Weren't you asking to make sure we consider the covariance? https://github.com/JuliaGraphics/ColorVectorSpace.jl/issues/126#issuecomment-599224928. The tensor product is the natural way to get the covariance---really, there shouldn't be "variance" and "covariance," there is just variance with different multiplication operators.

>  I think it's better to do lazy evaluation

Lazy works great if you're multiplying one RGB by another, but you need something to accumulate the (co)variance into. Typically, after combining the colors of 3 pixels you'll be at a rank-3 matrix, at which point you're better off using "dense" arithmetic.

Reply:
In the first place, is the tensor included in the target of ColorVectorSpace?
I think we should create a new package ColorTensorSpace.jl.:smile: Within the new package, we can feel more free to depend on the existing packages for tensors.

What we should do now is to clarify that multiple multiplication definitions can coexist, and I don't think we need to hurry to implement the tensor product. At least this PR can be split into more atomic changes.

Reply:
I agree that the tensor product is likely to see less use. If it seems premature to implement it in this PR, sure, I can split it out.

Reply:
I think the current implementation will help we understand the behavior. So, if we consider this to be an experimental feature (even if it seems premature), it may be good to leave the code for the time being. On the other hand, if we need more discussion, I think the separate PR is helpful.

Reply:
cf. https://discourse.julialang.org/t/ann-tensoralgebra-jl-taking-covariance-seriously/43587

Reply:
Once we start auto-differentiating with respect to RGB values, that could indeed be useful.